renku/cli/init.py

```diff
@@ -75,15 +75,40 @@
     https://github.com/SwissDataScienceCenter/renku-project-template@master
     ... OK
 
-    INDEX  ID              DESCRIPTION
-    -----  --------------  -------------------------------------------------
-    1  python-minimal  Basic Python Project: The simplest Python-based[...]
-    2  R-minimal       Basic R Project: The simplest R-based renku proj[...]
+    INDEX ID             DESCRIPTION                PARAMETERS
+    ----- -------------- -------------------------- ----------------------
+    1     python-minimal Basic Python Project:[...] description: proj[...]
+    2     R-minimal      Basic R Project: The [...] description: proj[...]
 
     Please choose a template by typing the index:
 
-Updating an existing project
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Provide parameters
+~~~~~~~~~~~~~~~~~-
+
+Some templates require parameters to properly initialize a new project. You
+can check them by listing the templates ``--list-templates``.
+
+To provide parameters, use the ``--parameter`` option and provide each
+parameter using ``--parameter "param1"="value1"``.
+
+.. code-block:: console
+
+    $ renku init --template-id python-minimal --parameter \
+    "description"="my new shiny project"
+
+    Initializing new Renku repository... OK
+
+If you don't provide the required parameters, the template will use an empty
+strings instead.
+
+.. note:: Every project requires a ``name`` that can either be provided using
+   ``--name`` or automatically taken from the target folder. This is
+   also considered as a special parameter, therefore it's automatically added
+   to the list of parameters forwarded to the ``init`` command. Providing a
+   different value for ``name`` through ``--parameter`` has no effect.
```

Comment:
We should hide the `name` parameter then. There is no point in showing it when users cannot set its value. Is this possible?

Reply:
The user can pass a custom `name`, otherwise `name = <current_folder_name>`.

This should clarify that passing a custom `--parameter name=something` has no effect because the parameter `name` is already automatically added from `--name`. It's a special case.

I fear it's not explained well :sweat_smile:  Please feel free to suggest something more understandable to users :slightly_smiling_face:

Reply:
It's just a bit confusing that `name` is shown in the parameters' list (`renku init -l`) but it cannot be set as a normal parameter. It would be ideal either to remove it from the parameters' list or remove `--name` and treat `name` as a normal parameter.

Reply:
oh yes, you are totally right! I will open a PR to update that in the templates repository. The original idea was slightly different, this is why `name` was added as a parameter there

Reply:
This should remove this confusing behavior SwissDataScienceCenter/renku-project-template#72

Once merged and tagged, no template will require `name` as a parameter anymore.

Reply:
Great! We can remove this line then: `Providing a different value for ``name`` through ``--parameter`` has no effect.`

Reply:
done :+1: